var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class QtBase(CMakePackage):
+    """Qt Base (Core, Gui, Widgets, Network, ...)"""
+
+    homepage = "https://www.qt.io"
+    url      = "https://github.com/qt/qtbase/archive/refs/tags/v6.2.3.tar.gz"
+    list_url = "https://github.com/qt/qtbase/tags"
+
+    maintainers = ['wdconinc', 'sethrj']
+
+    version('6.2.4', sha256='657d1405b5e15afcf322cc75b881f62d6a56f16383707742a99eb87f53cb63de')
+    version('6.2.3', sha256='2dd095fa82bff9e0feb7a9004c1b2fb910f79ecc6111aa64637c95a02b7a8abb')
+
+    generator = 'Ninja'
+
+    # Changing default to Release for typical use in HPC contexts
+    variant('build_type',
+            default='Release',
+            values=("Release", "Debug", "RelWithDebInfo", "MinSizeRel"),
+            description='CMake build type')
+
+    variant('accessibility', default=True, description='Build with accessibility support.')
+    variant('dbus', default=False, description='Build with D-Bus support.')
+    variant('examples', default=False, description='Build examples.')
+    variant('framework', default=False, description='Build as a macOS Framework package.')
+    variant('gtk', default=False, description='Build with gtkplus.')
+    variant('gui', default=True, description='Build the Qt GUI module and dependencies.')
+    variant('opengl', default=False, description='Build with OpenGL support.')
+    variant('shared', default=True, description='Build shared libraries.')
+    variant('sql', default=True, description='Build with SQL support.')
+    variant('ssl', default=True, description='Build with OpenSSL support.')
```

Comment:
Qt internally calls  this option `openssl` so line `96` really should be `openssl`. Since many other packages name the equivalent variant `openssl` let's rename it:
```diff
diff --git a/var/spack/repos/builtin/packages/qt-base/package.py b/var/spack/repos/builtin/packages/qt-base/package.py
index bcf0727cee..8361fcc0f9 100644
--- a/var/spack/repos/builtin/packages/qt-base/package.py
+++ b/var/spack/repos/builtin/packages/qt-base/package.py
@@ -33,7 +33,7 @@ class QtBase(CMakePackage):
     variant('opengl', default=False, description='Build with OpenGL support.')
     variant('shared', default=True, description='Build shared libraries.')
     variant('sql', default=True, description='Build with SQL support.')
-    variant('ssl', default=True, description='Build with OpenSSL support.')
+    variant('openssl', default=True, description='Build with OpenSSL support.')
     variant('widgets', default=True, description='Build with widgets.')

     depends_on('cmake@3.16:', type='build')
@@ -59,7 +59,7 @@ class QtBase(CMakePackage):
     depends_on('libxkbcommon')
     depends_on("libxml2")
     depends_on('libxrender')
-    depends_on('openssl', when='+ssl')
+    depends_on('openssl', when='+openssl')
     depends_on('pcre2+multibyte')
     depends_on("sqlite", when='+sql')
     depends_on("zlib")
@@ -93,7 +93,7 @@ def define_feature(variant):
             define_feature('dbus'),
             define_feature('framework'),
             define_feature('gui'),
-            define_feature('ssl'),
+            define_feature('openssl'),
             define_feature('widgets'),
             define_feature('sql'),
         ]
```
Currently `~ssl` results in an error:
```
2 errors found in build log:
     220    -- Performing Test HAVE_dtls
     221    -- Performing Test HAVE_dtls - Success
     222    -- Performing Test HAVE_ocsp
     223    -- Performing Test HAVE_ocsp - Success
     224    -- Performing Test HAVE_networklistmanager
     225    -- Performing Test HAVE_networklistmanager - Failed
  >> 226    CMake Error at cmake/QtBuildInformation.cmake:388 (message):
     227      Feature "openssl": Forcing to "ON" breaks its condition:
     228
     229          QT_FEATURE_openssl_runtime OR QT_FEATURE_openssl_linked
     230
     231      Condition values dump:
     232

     ...

     414        QT_FEATURE_openssl_runtime OR QT_FEATURE_openssl_linked
     415    Condition values dump:
     416        QT_FEATURE_openssl_runtime = "OFF"
     417        QT_FEATURE_openssl_linked = "OFF"
     418
     419
  >> 420    CMake Error at cmake/QtBuildInformation.cmake:80 (message):
     421      Check the configuration messages for an error that has occurred.
     422    Call Stack (most recent call first):
     423      cmake/QtBuildInformation.cmake:11 (qt_configure_print_summary)
     424      cmake/QtBuildInternals/QtBuildInternalsConfig.cmake:472 (qt_print_feature_summary)
     425      CMakeLists.txt:192 (qt_build_repo_end)
     426
```

Reply:
Actually, with openssl off, qt still looks for `libcrypto` but doesn't use it. Instead it looks for and links to a system GSSAPI package, which we don't want...

Maybe we need a variant `+network` that defaults to true, and require openssl if enabled?

Reply:
I think Qt has a mixture of [FEATURE_ssl](https://github.com/qt/qtbase/blob/dev/src/network/CMakeLists.txt#L322) and [FEATURE_openssl](https://github.com/qt/qtbase/blob/dev/src/plugins/tls/CMakeLists.txt#L5)...

Reply:
Oh dear. Do you have a preference on a network component variant versus twiddling with the ssl dependency? I generally think package variants should be defined by *capabilities* they enable rather than by *dependencies* they require, so `network` would be more appropriate.

Reply:
I think a `network` variant would make sense, in particular with since it's a fairly toplevel flag too, https://github.com/qt/qtbase/blob/27efadec226cabd14f417c9f44f86557ea88ec4c/src/CMakeLists.txt#L8. I think there are still plenty of places where the Qt6 docs are following the older configure options of Qt5 even when maybe the build system is more structured now.